internal/services/compute/client/helper.go

```diff
@@ -0,0 +1,29 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+)
+
+func (c *Client) CancelRollingUpgradesBeforeDeletion(ctx context.Context, resourceGroupName string, vmScaleSetName string) error {
+	future, err := c.VMScaleSetRollingUpgradesClient.Cancel(ctx, resourceGroupName, vmScaleSetName)
+
+	// If rolling upgrades haven't been run (when VMSS are just provisioned with rolling upgrades but no extensions, auto-scaling are run )
+	// we can not cancel rolling upgrades
+	// API call :: GET https://management.azure.com/subscriptions/{subId}/resourceGroups/{rgName}/providers/Microsoft.Compute/virtualMachineScaleSets/{vmSSName}/rollingUpgrades/latest?api-version=2021-07-01
+	// Azure API throws 409 conflict error saying "The entity was not found in this Azure location."
```

Comment:
@harshavmb I'm trying to reproduce this particular behaviour but am unable to. Could you provide some more details on what you mean by auto-scaling? As far as I can tell the resource doesn't expose any auto scaling properties.